pkg/workloads/cortex/client/cortex/client.py

```diff
@@ -33,7 +38,133 @@ def __init__(self, env: str):
         """
         self.env = env
 
+    # CORTEX_VERSION_MINOR x5
     def deploy(
+        self,
+        config: dict,
+        cls=None,
+        pip_dependencies=[],
+        conda_dependencies=[],
+        force: bool = False,
+        wait: bool = False,
+    ):
+        """
+        Deploy an API.
+
+        Args:
+            config: A dictionary defining a single Cortex API. Schema can be found here:
+                → Realtime API: https://docs.cortex.dev/v/master/deployments/realtime-api/api-configuration
+                → Batch API: https://docs.cortex.dev/v/master/deployments/batch-api/api-configuration
+                → Traffic Splitter: https://docs.cortex.dev/v/master/deployments/realtime-api/traffic-splitter
+            cls: A Cortex Predictor class implementation. Not required when deploying a traffic splitter.
+                → Realtime API: https://docs.cortex.dev/v/master/deployments/realtime-api/predictors
+                → Batch API: https://docs.cortex.dev/v/master/deployments/batch-api/predictors
+            force: Override any in-progress api updates.
+            wait: Streams logs until the APIs are ready.
+
+        Returns:
+            Deployment status, API specification, and endpoint for each API.
+        """
+        project_dir = Path.home() / ".cortex" / "deployments" / str(uuid.uuid4())
+        with util.open_tempdir(str(project_dir)):
+            if cls is None:
+                # for deploying a traffic splitter
+                return self.deploy_project(config, project_dir, wait=wait, force=force)
+
+            # Change if PYTHONVERSION changes
+            expected_version = "3.6"
+            actual_version = f"{sys.version_info.major}.{sys.version_info.minor}"
+            if actual_version < expected_version:
+                raise Exception("cortex is only supported for python versions >= 3.6")  # unexpected
+            if actual_version > expected_version:
+                is_python_set = False
+                for conda_dep in conda_dependencies:
+                    if "::python=" in conda_dep:
```

Comment:
Are we certain that python packages will always have the colons `::` in their package's name? What would happen if they weren't there? If my memory serves me well, you can also specify `python=3.6.x` in `conda-packages.txt` and that will still work, but it will install a different kind of Python (that we don't want) - I might be wrong though.

In such a case, there will be some sort of conflict when conda runs.